verification/utils/slices/slices.gobra

```diff
@@ -141,6 +141,39 @@ func NilAcc_Bytes() {
 	fold Bytes(nil, 0, 0)
 }
 
+ghost
+requires len(s1) > 0 || len(s2) > 0
+preserves Bytes(s1, 0, len(s1))
+preserves acc(Bytes(s2, 0, len(s2)), _)
+ensures s1 !== s2
+decreases
+func PermsImplyIneqWithWildcard(s1 []byte, s2 []byte) {
+	unfold Bytes(s1, 0, len(s1))
+	unfold acc(Bytes(s2, 0, len(s2)), _)
+	if len(s1) > 0 && len(s2) > 0 {
+		assert &s1[0] != &s2[0]
+	}
+	fold Bytes(s1, 0, len(s1))
+	fold acc(Bytes(s2, 0, len(s2)), _)
+}
+
+ghost
+requires 0 < p
+requires len(s1) > 0 || len(s2) > 0
+preserves Bytes(s1, 0, len(s1))
+preserves acc(Bytes(s2, 0, len(s2)), p)
+ensures s1 !== s2
+decreases
+func PermsImplyIneq(s1 []byte, s2 []byte, p perm) {
```

Comment:
Unfortunately, we need two versions of this lemma because `PermsImplyIneqWithWildcard() `cannot guarantee that it returns the same amount of permissions for `s2`. Is there a way to invoke `PermsImplyIneq(s1, s2, _)`, where _ serves as a wildcard permission?

Reply:
That's  to be expected. This could be in principle done with permission introspection (i.e., using the `perm(x)` expressions from Viper that return the permission amount held for `x`), but it is likely we will never have this in Gobra, as this is too low-level of a primitive that is easy to abuse. It's fine to have the two theorems if they are both needed